opta/nice_subprocess.py

```diff
@@ -24,19 +24,24 @@
 
 
 def log_to_datadog(msg: str, severity: str) -> None:
-    msg = ansi_scrub(msg)
-    if hasattr(sys, "_called_from_test") or VERSION == DEV_VERSION or not VERSION:
-        if os.environ.get("OPTA_DEBUG", "") == "DATADOG_LOCAL":
-            print("Would have logged this string to DD:\n")
-            print(">>>>>>>>>>>>>>>>>>>>Datadog log start")
-            print(msg)
-            print("<<<<<<<<<<<<<<<<<<<<Datadog log end")
-        dd_handler.setLevel(logging.CRITICAL)
+    try:
+        msg = ansi_scrub(msg)
+        if hasattr(sys, "_called_from_test") or VERSION == DEV_VERSION or not VERSION:
+            if os.environ.get("OPTA_DEBUG", "") == "DATADOG_LOCAL":
+                print("Would have logged this string to DD:\n")
+                print(">>>>>>>>>>>>>>>>>>>>Datadog log start")
+                print(msg)
+                print("<<<<<<<<<<<<<<<<<<<<Datadog log end")
+            dd_handler.setLevel(logging.CRITICAL)
 
-    if severity == "ERROR":
-        nice_logger.error(msg)
-    else:
-        nice_logger.info(msg)
+        if severity == "ERROR":
+            nice_logger.error(msg)
+        else:
+            nice_logger.info(msg)
+    except:  # nosec   # noqa: E722
+        # Logging error to datadog SaaS, happens,
+        # not the end of world, silently fail
+        pass
```

Comment:
I wonder if we would want to see this error _only_ if `OPTA_DEBUG` is set.

Reply:
> I wonder if we would want to see this error _only_ if `OPTA_DEBUG` is set.

The underlying issue that casued this hotfix was that datadog endpoint wasnt reachable by the user who reported the problem; in all debug runs we internally usually don't send logs to datadog.